src/GafferScene/ShaderTweaks.cpp

```diff
@@ -316,8 +396,84 @@ bool ShaderTweaks::applyTweaks( IECoreScene::ShaderNetwork *shaderNetwork, Tweak
 				{
 					throw IECore::Exception( fmt::format( "Cannot apply tweak to \"{}\" : Mode must be \"Replace\" when inserting a connection", name ) );
 				}
+
 				const auto inputParameter = ShaderNetworkAlgo::addShaders( shaderNetwork, inputNetwork );
 				shaderNetwork->addConnection( { inputParameter, parameter } );
+
+				// TODO - it would be substantially more efficient to search for and process tweak sources
+				// just in `inputNetwork` before merging it to `shaderNetwork` ... but this would require
+				// dealing with weird connections where the input node handle is relative to `shaderNetwork`,
+				// but the output handle is relative to `inputNetwork`. This gets very confusing if there are
+				// nodes in the two networks with the same name, which get uniquified during addShaders.
+				// Doing this after merging simplifies all that.
+				std::vector<IECore::InternedString> shadersToDelete;
+				for( const auto &i : shaderNetwork->shaders() )
+				{
+					if( i.second->getName() == ShaderTweakProxy::shaderTweakProxyIdentifier() )
+					{
+						shadersToDelete.push_back( i.first );
+
+						ShaderNetwork::ConnectionRange range = shaderNetwork->outputConnections( i.first );
+						const std::vector<ShaderNetwork::Connection> outputConnections( range.begin(), range.end() );
+
+						for( const auto &c : outputConnections )
+						{
+							ShaderNetwork::Parameter dest = c.destination;
+
+							StringData *sourceNodeData = nullptr;
+							try
+							{
+								sourceNodeData = IECore::runTimeCast< StringData >(
+									i.second->parameters().at( "shaderTweakProxySourceNode" ).get()
+								);
+							}
+							catch( ... )
+							{
+							}
+
+							if( !sourceNodeData )
+							{
+								throw IECore::Exception( "Cannot find source node parameter on ShaderTweakProxy" );
+							}
+							const std::string sourceNode = sourceNodeData->readable();
+
+							shaderNetwork->removeConnection( c );
+							removedConnections = true;
+
+							if( sourceNode == "" )
+							{
+								if( originalInput )
+								{
+									shaderNetwork->addConnection( { originalInput, dest } );
+								}
+								else
+								{
+									const IECoreScene::Shader *proxyConnectedShader = shaderNetwork->getShader( dest.shader );
+									if( !proxyConnectedShader )
+									{
+										throw IECore::Exception( fmt::format( "ShaderTweakProxy connected to non-existent shader \"{}\"", dest.shader.string() ) );
+									}
+
+									// Regular tweak
+									auto modifiedShader = modifiedShaders.insert( { dest.shader, nullptr } );
+									if( modifiedShader.second )
+									{
+										modifiedShader.first->second = proxyConnectedShader->copy();
+									}
+
+									const IECore::Data *origDestParameter = modifiedShader.first->second->parameters().at(dest.name).get();
```

Comment:
I don't recall how I achieved it, but I got a rather non-descriptive exception thrown from `at()` while playing with this, and I presume it came from here. `parametersData()->member( /* throwExceptions = */ true )` would be an improvement because it would include the `dest.name` in the error message.

Reply:
Looks like the call to `at()` is still there?